src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_parquet.rs

```diff
@@ -332,8 +333,14 @@ impl BlockBuilderTrait for ParquetBlockBuilder {
     fn deserialize(&mut self, mut batch: Option<RowGroupInMemory>) -> Result<Vec<DataBlock>> {
         if let Some(rg) = batch.as_mut() {
             let chunk = rg.get_arrow_chunk()?;
-            let block = DataBlock::from_chunk(&self.ctx.schema, &chunk)?;
-            Ok(vec![block])
+            // Using `block_compact_thresholds` is enough here.
```

Comment:
I don't think we should add functions in `DataBlock`. this crate will be removed later.

Maybe this will be better:



```
 let block = DataBlock::from_chunk(&self.ctx.schema, &chunk)?; 
            let blocks = (0..block.num_rows()).step_by(max_rows_per_block).for_each(|index| {
                if index + max_rows_per_block > block.num_rows() {
                    block.slice(index, max_rows_per_block)
                } else {
                    block.slice(index, max_rows_per_block - index)
                }
            }).collect();
            Ok(blocks)
```

Reply:
So the `DataBlock::from_chunk` will also be removed?